distributed/scheduler.py

```diff
@@ -9046,11 +9055,12 @@ def transition(
 
 
 def _materialize_graph(
-    graph: HighLevelGraph, global_annotations: dict[str, Any]
+    graph: HighLevelGraph, global_annotations: dict[str, Any], validate: bool
 ) -> tuple[dict[Key, T_runspec], dict[Key, set[Key]], dict[str, dict[Key, Any]]]:
     dsk = ensure_dict(graph)
-    for k in dsk:
-        validate_key(k)
+    if validate:
```

Comment:
There are a few seemingly unrelated changes in this file. Let's extract them to a separate PR.

Reply:
https://github.com/dask/distributed/pull/8822